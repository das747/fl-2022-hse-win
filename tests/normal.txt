&A& -> &B& &C&
&B& -> @b@
=> &S& -> &A& &D&
&D& -> @d@
&C& -> @c@