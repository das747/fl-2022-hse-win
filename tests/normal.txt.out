LexToken(NON_TERM,'A',25,0)
LexToken(RULE,'->',25,4)
LexToken(NON_TERM,'B',25,7)
LexToken(NON_TERM,'C',25,11)
LexToken(NON_TERM,'B',26,15)
LexToken(RULE,'->',26,19)
LexToken(TERM,'b',26,22)
LexToken(START,'=>',27,26)
LexToken(NON_TERM,'S',27,29)
LexToken(RULE,'->',27,33)
LexToken(NON_TERM,'A',27,36)
LexToken(NON_TERM,'D',27,40)
LexToken(NON_TERM,'D',28,44)
LexToken(RULE,'->',28,48)
LexToken(TERM,'d',28,51)
LexToken(NON_TERM,'C',29,55)
LexToken(RULE,'->',29,59)
LexToken(TERM,'c',29,62)

Start non-terminal:
	'S'

Non-terminals:
	'C'
	'A'
	'D'
	'S'
	'B'

Terminals:
	'd'
	'b'
	'c'

Rules:
	'A' -> 'B' 'C'
	'B' -> 'b'
	'S' -> 'A' 'D'
	'D' -> 'd'
	'C' -> 'c'

Grammar is in Chomsky normal form
